### PR TITLE
docs: Added getting-started & custom sphinx extension documentation

### LINKED ISSUE #245: document current setup
Especially since we have some new people joining our ranks, but also in general, we need to describe how the current tooling is set up.

Specifically (to be extended):
* How is docs-as-code set up (bazel integration, esbonio setup, score_metamodel, _tooling directory, ...)
* What is the goal for multi repo docs output? multiple webpages? single page? compare https://github.com/eclipse-score/score/issues/218

Current approach: so far we document such things in README.md files in the respective directories.

### PR DESCRIPTION
**Pull request by MaximilianSoerenPollak**
Added documentation aiding in explaining _tooling structure. 
Added documentation to help develop new custom sphinx extensions

Hoping this gives a good basis to build on top of for future documentation and guides.

closes #245

### CODE REVIEW

File: docs/_tooling/README.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+#### Documentation Development
+```bash
+# Incremental documentation build (recommended during development)
+bazel run //docs:incremental
+
+# Full documentation build
+bazel build //docs:docs
+```
+
+#### Testing
```
Comment (AlexanderLanin) [inline docs/_tooling/README.md:81]: The IDE integration is missing. I feel this is more important than command line commands.

File: docs/_tooling/docs/extension_guide.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+As we want to ensure code quality, testing is an integral part of the development process.
+We perform testing with unit tests as well as integration tests that validate the full extension.
+
+> Tests should cover both the happy path and edge cases.
+
+### Where to Place Your Test Code
+
+```
+_tooling
+��� extensions
```
Comment (AlexanderLanin) [inline docs/_tooling/docs/extension_guide.md:91]: `ÃÄÄ` ??

File: docs/_tooling/docs/extension_guide.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+#### What Do You Need?
+
+To create a Sphinx testing app, you need the same components as a normal Sphinx app:
+- RST file to build
+- `conf.py` file
+- Source, conf, and build directories
+
+In addition, you can provide anything else that you might need to test your specific extension.
+
+#### Building the Fixtures
```
Comment (AlexanderLanin) [inline docs/_tooling/docs/extension_guide.md:119]: The below code is rather low level and I would hope that we find a more convenient solution over time. I don't particularly like having code in a .md file. Can't we just point at some example? Because that example will be kept up to date with whatever we improve. This .md will not.

**Review by AlexanderLanin — CHANGES_REQUESTED** [review]

File: docs/_tooling/README.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+- *With Devcontainer (VSCode)*
+    1. Click the `reopen current folder in dev container` prompt
+        -  If no prompt appears: `ctrl+shift+p` => `Dev Containers: Reopen in Containers`
+
+
+## Development Environment Requirements
+
+- **Operating System**: Linux (required)
+- **Core Tools**:
+  - Bazel 7.4.0 (via Bazelisk)
```
Comment (hoe-jo) [inline docs/_tooling/README.md:29]: Do we really want to specify the tool versions in this guide? It will get soon outdated ;)

File: docs/_tooling/README.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+│   └── score_metamodel/
+│       ├── checks/  # Sphinx-needs validation
+│       └── tests/   # Extension test suite
+└── templates/        # Documentation templates
+```
+
+## Development Workflow
+
+
+#### Documentation Development
```
Comment (hoe-jo) [inline docs/_tooling/README.md:72]: Do you really want to duplicate the commands here or just put a link here? They are already mentioned in the project readme.md?

File: docs/_tooling/README.md (hunk 1 above)
Comment (hoe-jo) [inline docs/_tooling/README.md:81]: Testing is also mentioned in the project readme.me. Or do you rather want to specify which targets there are and if they are executed in the CI?

File: docs/_tooling/docs/extension_guide.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+```python
+    try:
+        app.build()
+        html_content = (app.outdir / "index.html").read_text() 
+        assert "<td><p>John Doe</p></td>" in html_content
+        assert '<div class="line">approver_1</div>' in html_content
+        # ... other assertions 
+```
+
+### How to Add Tests to Bazel
```
Comment (hoe-jo) [inline docs/_tooling/docs/extension_guide.md:268]: Let´s maybe put this in the documentaion in a Bazel Tutorial?

File: docs/_tooling/README.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
+# Score Project Tooling Development Guide
+
+*This document is meant for *developers* of the `_tooling` aspect of the score repository.*
```
[the remark this answers is not in the document]
Comment (MaximilianSoerenPollak) [reply docs/_tooling/README.md:3]: Indeed, we probably should only use it for the 'bazel' usecase to avoid confusion and keep consitency.

File: docs/_tooling/README.md (hunk 4 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/README.md:29]: You're right there, I think just pointing to the `requirements_lock` would be a nicer way.

File: docs/_tooling/README.md (hunk 5 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/README.md:72]: Removed duplication and just pointed towards the appropiate README

File: docs/_tooling/README.md (hunk 1 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/README.md:81]: Removed duplication and just pointed towards the appropiate README

File: docs/_tooling/docs/extension_guide.md (hunk 3 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/docs/extension_guide.md:119]: That is true, I would also like to work on a 'general' solution (like the CheckLogger) just some generic SphinxTestApp we can import from the tests init file that can be used anywhere needed. 

The mentioned example here is not yet merged into main it's the `source_code_linker` #427 . But yes, just pointing towards that might be a better idea than having such low level explanations here.
I just wanted to provide more information than the sphinx-testing as there is almost 0, and they only point towards the implementation

File: docs/_tooling/docs/extension_guide.md (hunk 2 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/docs/extension_guide.md:91]: Ahh, great I must have pasted the UNICODE instead of actually the tree. Will correct this.

File: docs/_tooling/docs/extension_guide.md (hunk 6 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/docs/extension_guide.md:268]: Removed this and added a hint to look at the pytest readme

File: docs/_tooling/README.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+- **Operating System**: Linux (required)
+- **Core Tools**:
+  - Bazel 
+  - Python 
+  - Git
+  - **VSCode** (Optional)
+    - Several integrations and guides are development primarily with VS Code in mind.
+
+For current versions used please see the [requirement lockfile](/docs/_tooling/requirement_lock.txt)
```
Comment (AlexanderLanin) [inline docs/_tooling/README.md:35]:
```suggestion

```

That's not a "Development Environment Requirements" people need to install before using score.

File: docs/_tooling/docs/extension_guide.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
+# Guide to Creating a Sphinx Extension
```
Comment (AlexanderLanin) [inline docs/_tooling/docs/extension_guide.md:1]: Shouldn't this file be under `docs/extensions`?

File: docs/_tooling/README.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+│   └── score_metamodel/
+│       ├── checks/  # Sphinx-needs validation
+│       └── tests/   # Extension test suite
+└── templates/        # Documentation templates
+```
+
+
+Find all important bazel commands in the [project README](/README.md)
+
+Find everything related to testing and how to add your on test suite [here](/tools/testing/pytest/README.md)
```
Comment (dcalavrezo-qorix) [inline docs/_tooling/README.md:75]: I don't think the links will work maybe ../../tools/testing...?

File: docs/_tooling/docs/extension_guide.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+#### What Do You Need?
+
+To create a Sphinx testing app, you need the same components as a normal Sphinx app:
+- RST file to build
+- `conf.py` file
+- Source, conf, and build directories
+In addition, you can provide anything else that you might need to test your specific extension.
+
+
+For examples on how to use and implement the sphinx testing app, you can check out the [source code linker](link here)
```
Comment (AlexanderLanin) [inline docs/_tooling/docs/extension_guide.md:121]: "link here"

**Review by AlexanderLanin — CHANGES_REQUESTED** [review]

**Review by dcalavrezo-qorix — APPROVED** [review]

File: docs/_tooling/docs/extension_guide.md (hunk 11 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/docs/extension_guide.md:121]: Waiting for source code linker to be merged to add the link then there.

File: docs/_tooling/docs/extension_guide.md (hunk 9 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/docs/extension_guide.md:1]: you mean in `docs/extensions/docs` ? 

Just putting it in `extensions` doesn't seem good as other ones might join it then it will be a lot.

Comment (AlexanderLanin) [reply docs/_tooling/docs/extension_guide.md:1]: It seems so basic, that it should even be in `docs/_tooling/extensions/README.md` ?

Comment (MaximilianSoerenPollak) [reply docs/_tooling/docs/extension_guide.md:1]: It might should be in there ye. Should I just move this whole doc to there then?

File: docs/_tooling/README.md (hunk 8 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/README.md:35]: Should I just drop the `requirements` ? To make it more clear / easier?

File: docs/_tooling/README.md (hunk 10 above)
Comment (MaximilianSoerenPollak) [reply docs/_tooling/README.md:75]: I just tried, it does indeed work :), so should be fine.

Comment (MaximilianSoerenPollak) [reply docs/_tooling/README.md:75]: Although, this was on a server that was started from the `score` folder, so that might be the reason this works.

Comment (AlexanderLanin) [reply docs/_tooling/README.md:74]: I don't understand why these links work, but they do work.

Rendered markdown at https://github.com/eclipse-score/score/blob/3882497aa246a9bfc8769c5a6380735ca24a5557/docs/_tooling/README.md
